Explorer/Assets/DCL/WebRequests/Analytics/Plugin/ShowWebRequestsAnalyticsSystem.cs

```diff
@@ -1,58 +1,75 @@
-            DebugWidgetBuilder widget = debugContainerBuilder.AddWidget("Web Requests")
-                                                             .SetVisibilityBinding(visibilityBinding = new DebugWidgetVisibilityBinding(true));
-
-            foreach (Type requestType in requestTypes)
-            {
-                var binding = new ElementBinding<ulong>(0);
-                ongoingRequests[requestType] = binding;
-
-                widget.AddMarker(requestType.Name, binding, DebugLongMarkerDef.Unit.NoFormat);
-            }
-        }
-
-        protected override void Update(float t)
-        {
-            if (visibilityBinding.IsExpanded && lastTimeSinceMetricsUpdate > THROTTLE)
-            {
-                lastTimeSinceMetricsUpdate = 0;
-
-                foreach (Type requestType in requestTypes)
-                    ongoingRequests[requestType].Value = (ulong)webRequestsAnalyticsContainer.GetMetric(requestType);
-            }
-
-            lastTimeSinceMetricsUpdate += t;
-        }
-    }
-}
+﻿using Arch.Core;
+using Arch.SystemGroups;
+using Arch.SystemGroups.DefaultSystemGroups;
+using DCL.DebugUtilities;
+using DCL.DebugUtilities.UIBindings;
+using ECS.Abstract;
+using System;
+using System.Collections.Generic;
+
+namespace DCL.WebRequests.Analytics
+{
+    [UpdateInGroup(typeof(PresentationSystemGroup))]
+    public partial class ShowWebRequestsAnalyticsSystem : BaseUnityLoopSystem
+    {
+        private const float THROTTLE = 0.1f;
+
+        private readonly IWebRequestsAnalyticsContainer webRequestsAnalyticsContainer;
+        private readonly Type[] requestTypes;
+
+        private readonly Dictionary<Type, Dictionary<string, ElementBinding<ulong>>> ongoingRequests = new ();
+        private readonly DebugWidgetVisibilityBinding visibilityBinding;
+
+        private float lastTimeSinceMetricsUpdate;
+
+        internal ShowWebRequestsAnalyticsSystem(World world,
+            IWebRequestsAnalyticsContainer webRequestsAnalyticsContainer,
+            IDebugContainerBuilder debugContainerBuilder,
+            Type[] requestTypes) : base(world)
+        {
+            this.webRequestsAnalyticsContainer = webRequestsAnalyticsContainer;
+            this.requestTypes = requestTypes;
+
+            DebugWidgetBuilder widget = debugContainerBuilder.AddWidget("Web Requests")
+                                                             .SetVisibilityBinding(visibilityBinding = new DebugWidgetVisibilityBinding(true));
+
+            foreach (Type requestType in requestTypes)
+            {
+                var bindings = new Dictionary<string, ElementBinding<ulong>>(0);
+                var metrics = webRequestsAnalyticsContainer.GetTrackedMetrics();
+                foreach (var metric in metrics)
+                {
+                    bindings.Add(metric.Name, new ElementBinding<ulong>(0));
+                    widget.AddMarker(requestType.Name + "-" +  metric.Name, bindings[metric.Name], DebugLongMarkerDef.Unit.NoFormat);
```

Comment:
Do you mind extending `DebugLongMarkerElement` a bit so it can `Normalize` the value even if `NoFormat` is supplied? Similar to `BytesFormatter.Normalize` but without units (just generalize that function a bit more).
Otherwise, those values will be hard to read as their magnitude grows

Reply:
I have updated to include the ability to specify the format